pkg/registry/core/rest/storage_core.go

```diff
@@ -372,76 +446,119 @@ func (c LegacyRESTStorageProvider) newServiceIPAllocators() (registries kubernet
 				etcd, err := serviceallocator.NewEtcd(mem, "/ranges/secondaryserviceips", serviceStorageConfig.ForResource(api.Resource("serviceipallocations")))
 				if err != nil {
 					return nil, err
 				}
-				registries.SecondaryServiceClusterIPRegistry = etcd
+				registries.secondaryClusterIP = etcd
 				return etcd, nil
 			})
 			if err != nil {
-				return kubernetesservice.RangeRegistries{}, nil, nil, nil, fmt.Errorf("cannot create cluster secondary IP allocator: %v", err)
+				return rangeRegistries{}, nil, nil, nil, fmt.Errorf("cannot create cluster secondary IP allocator: %v", err)
 			}
 		} else {
 			networkingv1alphaClient, err := networkingv1alpha1client.NewForConfig(c.LoopbackClientConfig)
 			if err != nil {
-				return kubernetesservice.RangeRegistries{}, nil, nil, nil, err
+				return rangeRegistries{}, nil, nil, nil, err
 			}
-			secondaryClusterIPAllocator, err = ipallocator.NewIPAllocator(&c.SecondaryServiceIPRange, networkingv1alphaClient, c.Informers.Networking().V1alpha1().IPAddresses())
+			secondaryClusterIPAllocator, err = ipallocator.NewIPAllocator(&c.Services.SecondaryClusterIPRange, networkingv1alphaClient, c.Informers.Networking().V1alpha1().IPAddresses())
 			if err != nil {
-				return kubernetesservice.RangeRegistries{}, nil, nil, nil, fmt.Errorf("cannot create cluster secondary IP allocator: %v", err)
+				return rangeRegistries{}, nil, nil, nil, fmt.Errorf("cannot create cluster secondary IP allocator: %v", err)
 			}
 		}
 		secondaryClusterIPAllocator.EnableMetrics()
 		clusterIPAllocators[secondaryClusterIPAllocator.IPFamily()] = secondaryClusterIPAllocator
 	}
 
-	nodePortAllocator, err = portallocator.New(c.ServiceNodePortRange, func(max int, rangeSpec string, offset int) (allocator.Interface, error) {
+	nodePortAllocator, err = portallocator.New(c.Services.NodePortRange, func(max int, rangeSpec string, offset int) (allocator.Interface, error) {
 		mem := allocator.NewAllocationMapWithOffset(max, rangeSpec, offset)
 		// TODO etcdallocator package to return a storage interface via the storageFactory
 		etcd, err := serviceallocator.NewEtcd(mem, "/ranges/servicenodeports", serviceStorageConfig.ForResource(api.Resource("servicenodeportallocations")))
 		if err != nil {
 			return nil, err
 		}
-		registries.ServiceNodePortRegistry = etcd
+		registries.nodePort = etcd
 		return etcd, nil
 	})
 	if err != nil {
-		return kubernetesservice.RangeRegistries{}, nil, nil, nil, fmt.Errorf("cannot create cluster port allocator: %v", err)
+		return rangeRegistries{}, nil, nil, nil, fmt.Errorf("cannot create cluster port allocator: %v", err)
 	}
 	nodePortAllocator.EnableMetrics()
 
 	return
 }
 
-func (p LegacyRESTStorageProvider) GroupName() string {
+var _ genericapiserver.PostStartHookProvider = &legacyProvider{}
+
+func (p *legacyProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	return "start-service-storage-controllers", func(context genericapiserver.PostStartHookContext) error {
```

Comment:
does this not have to show up in the `test/e2e/apimachinery/health_handlers.go`?

Reply:
checking. Good point.

Reply:
added it to the list there. Note that the `requiredHealthzChecks` are only required, not exhaustive. That's why we didn't get an error.